web/modules/custom/farm_nfa/js/farmOS.map.behaviors.farm_nfa_plan_locations_layers.js

```diff
@@ -1,12 +1,73 @@
 (function () {
   farmOS.map.behaviors.farm_nfa_plan_locations_layers = {
-    attach: function (instance) {
+    attach: async function (instance) {
       var url = new URL('/nfa-assets/geojson/' + instance.farmMapSettings.plan, window.location.origin + drupalSettings.path.baseUrl)
       var newLayer = instance.addLayer('geojson', {
         title: Drupal.t('Locations'),
         url,
         color: 'orange',
       })
+
+      // Add a layer for fire alerts in the GFW plan tab
+      let gfwRegex = /\/plan\/\d+\/gfw/;
```

Comment:
I still don't clearly understand what this does? could you walk me through it on slack? thanks!

Reply:
surely, to brief though what this is apparently doing is the map is basically been rendered at many routes but we don't want
the fire and deforestation layer to get added on every route, we want it to get added just on the GFW section, so basically to check that i am using a regex that , matches the GFW route and runs the code to add the fires and deforestation layer just for the GFW section. hope this clears up this a bit
happy to sync on slack too to discuss this further !

Reply:
@Yash621 please check how to do this directly on PHP rather than js, for example https://www.computerminds.co.uk/articles/context-aware-blocks

Reply:
hey @pcambra instead of context-aware-blocks, I am thinking of passing an attribute like `'id' => 'gfw-map-component'` in the `forestPlanGfwForm.php` and checking the existence of the `gfw map` based on that in js file, will this approch be viable, or will using contextual blocks be a better option instead ?